SEED-platform/seed
# #2079 refactor(tasks): run geocoding task in celery

#### Any background context you want to provide?
When importing a large file, you can experience timeouts when starting the final saving process by clicking "Save Mappings". This was happening because in the `start_system_matching_and_geocoding` endpoint, we would start the geocoding task, then block until we got the result, before finally kicking off the matching task and returning the progress data. If the geocoding task took a while to run, or was blocked itself because the worker(s) were already occupied (data quality checks are probably still running for a large file) then it's going to block for a long time. In our case, ELB would timeout after 60s and return an empty HTTP response (which cloudflare doesn't like, and ends up giving a 520 response).

#### What's this PR do?
This address the timeout by not blocking on the geocoding task. Specifically, I chained together the geocoding and matching tasks because the should always happen together.

#### How should this be manually tested?
Because I created a new task for combining the previous tasks, you need to test a few things. It touches geocoding as well as the mapping of additional BuildingSync models (which occurs instead of the buildings matching).
Please test:
- upload a spreadsheet that won't be geocoded (example-pm-properties) and verify it was uploaded properly
- upload a spreadsheet that will be geocoded, and verify it was geocoded successfully (must have valid MapQuest API key in org settings)
- upload a spreadsheet that will be geocoded, but have an INVALID MapQuest API key in org settings; you should get an error reporting a bad or overused key
- upload a BSync file that has measure data (buildingsync_ex01_measures.xml), and verify the Measures were successfully created by checking the property detail page and scrolling to the Measures section

#### What are the relevant tickets?
#2079
